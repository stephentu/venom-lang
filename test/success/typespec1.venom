class A{T}
  attr elem::T
  def self(elem::T) = self.elem = elem; end
  def overrideMe() -> T = elem; end
end

class B{TT} <- A{TT}
  def self(elem::TT) : super(elem) = end
end

class C <- B{string}
  def self(elem::string) : super(elem) = end
end

class D{TTT} <- C
  def self(elem::string) : super(elem) = end
  def foo(x::TTT) = print(x); end
  def overrideMe() -> string = super.overrideMe(); end
end

d = D{int}("foo");
d.foo(12345);
print(d.elem);
print(d.overrideMe());
