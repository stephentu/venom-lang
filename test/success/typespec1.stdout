12345
foo
foo
